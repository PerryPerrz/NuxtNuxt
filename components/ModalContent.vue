<template>
    <div>
        <h2>{{ title }}</h2>
        <p>{{ content }}</p>
        <button @click="$emit('close')" data-type="close">Close</button>
    </div>
</template>

<script setup>
const props = defineProps({
    title: String,
    content: String,
});

const emit = defineEmits(['close']);
</script>

<style scoped>
    .modal h2 {
        margin-top: 0;
    }

    .modal p {
        margin-bottom: 0;
    }

    .modal button {
        background-color: transparent;
        border: none;
        cursor: pointer;
        position: absolute;
        right: 0.5rem;
        top: 0.5rem;
        color: lightpink;

        &:hover {
            color: red;
        }
    }
</style>