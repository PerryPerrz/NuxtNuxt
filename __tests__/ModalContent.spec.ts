import ModalContent from '../components/ModalContent.vue';
import { describe, test, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';

describe('ModalContent', () => {
    test('ModalContent is defined', () => {
        expect(ModalContent).toBeDefined();
    });
});

describe('ModalContent functions', () => {
    const wrapper = mount(ModalContent);

    it('should render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render correctly bis', () => {
        expect(wrapper.html()).toContain('Close');
    })
});

describe('ModalContent props', () => {
    const wrapper = mount(ModalContent, {
        props: {
            title: 'Test title',
            content: 'Test content'
        }
    });

    it('should render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render correctly bis', () => {
        expect(wrapper.html()).toContain('Test title');
        expect(wrapper.html()).toContain('Test content');
    })
})

describe('ModalContent emits', () => {
    const wrapper = mount(ModalContent);

    it('should emit close event', async () => {
        await wrapper.find('[data-type="close"]').trigger('click');
        expect(wrapper.emitted()).toHaveProperty('close');
    })
})

