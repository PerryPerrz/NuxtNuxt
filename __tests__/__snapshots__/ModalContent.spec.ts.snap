// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ModalContent functions > should render correctly 1`] = `
"<div data-v-3a899d1d="">
  <h2 data-v-3a899d1d=""></h2>
  <p data-v-3a899d1d=""></p><button data-v-3a899d1d="" data-type="close">Close</button>
</div>"
`;

exports[`ModalContent props > should render correctly 1`] = `
"<div data-v-3a899d1d="">
  <h2 data-v-3a899d1d="">Test title</h2>
  <p data-v-3a899d1d="">Test content</p><button data-v-3a899d1d="" data-type="close">Close</button>
</div>"
`;
